src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullBeforeInstanceOfCheck.java

```diff
@@ -0,0 +1,150 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+/**
+ * <p>
+ * Checks if unnecessary null check is used before instanceof expression.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code tokens} - tokens to check
+ * Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_IF">
+ * LITERAL_IF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NullBeforeInstanceOf&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *     if (myObj != null &amp;&amp; myObj instanceof Object) { // violation
+ *
+ *     }
+ *
+ *     if (myObj != null) { // violation
+ *         if (myObj instanceof Object) {
+ *
+ *         }
+ *     }
+ *
+ *    if ((myObj != null) &amp;&amp; (myObj instanceof Object)) {}  // violation
+ * }
+ * </pre>
+ *
+ * @since 8.33
+ */
+@StatelessCheck
+public class NullBeforeInstanceOfCheck extends AbstractCheck {
```

Comment:
From https://github.com/checkstyle/checkstyle/issues/6943#issuecomment-636357136

Please rename check to `NullTestAroundInstanceOf `

Reply:
done